src/main/java/com/wire/xenon/backend/GenericMessageProcessor.java

```diff
@@ -174,17 +188,21 @@ public boolean process(UUID from, String clientId, UUID convId, String time, Mes
         }
 
         if (asset != null) {
-            Logger.debug("Asset: msgId: %s hasOriginal: %s, hasUploaded: %s, hasPreview: %s",
+            Logger.debug("eventId: %s, msgId: %s hasOriginal: %s, hasUploaded: %s, hasPreview: %s",
+                    eventId,
                     messageId,
                     asset.hasOriginal(),
                     asset.hasUploaded(),
                     asset.hasPreview());
 
             if (asset.hasPreview()) {
                 ImageMessage msg = new ImageMessage(messageId, convId, clientId, from);
-                handleVideoPreview(asset.getPreview(), msg, time);
+                msg.setEventId(eventId);
+
+                handleVideoPreview(asset.getPreview(), msg, time);  //todo is this legacy?
```

Comment:
`handleVideoPreview` while processing an Image?

Reply:
Yeh. I wrote there some todo. I am not sure why video preview is sent differently by the clients. I think I had an issue with this in Recording bot and I think Web App sends video differently than mobile clients. We can try to remove this code and see what/if something breaks